### ISSUE TITLE
Save PrettyPageHandler instance in database.

### ISSUE THREAD

**Issue by aidanthewiz** [filp/whoops#669]
I am trying to save an instance of PrettyPageHandler to view at a later date because it has more useful information in the tables but the only solution I have found so far is to extend the class and save the HTML output to the database. 

Obviously this is not the best solution as the actual data that is being loaded into the view is all that needs to be saved. Is there a method I am missing to export the data needed to load back into the PrettyPageHandler, or maybe that's something we could add? Here is my current code that saves the HTML to the database:

```php
class DatabaseHandler extends \Whoops\Handler\PrettyPageHandler {
	public function handle() {
		parent::handle();
		$this->CI =& get_instance();

		$this->CI->db->insert('error_log', [
			'whoops_instance' => ob_get_clean()
		]);
	}
}

class WhoopsHook
{
	public function initWhoops()
	{
		$whoops = new \Whoops\Run;
		if (ENVIRONMENT !== 'production')
		{
			if (\Whoops\Util\Misc::isAjaxRequest())
			{
				$jsonHandler = new \Whoops\Handler\JsonResponseHandler;
				$jsonHandler->addTraceToOutput(true);
				$whoops->pushHandler($jsonHandler);
			} else {
				$whoops->pushHandler(new \Whoops\Handler\PrettyPageHandler);
			}
		}

		$whoops->pushHandler(new DatabaseHandler);

		$whoops->register();
	}
}
```

**Comment by staabm** [comment]
I guess saving the html (which is self contained: it contains all information, css, js required) would be your best bet.

In case you would store the raw information you could easily miss something. Also if whoops in the futuree would take some new parameters into account you would also have to add them afterwards.

Maybe we could add som serializing hooks to make this case easier...

**Comment by aidanthewiz** [comment]
Yeah, for now saving the HTML seemed like the best bet but I'd rather not have it in my database so I might have it save to AWS S3 bucket. 

Best case scenario I was wondering about the possibility of doing something like the \Whoops\Handler\Handler setRun(), setInspector(), and setException() and corresponding get methods, but for the Environment data and other tables that the PrettyPageHandler has. Obviously I could save the $_SERVER, $_POST, etc variables along with the Exception to the database, but if someone includes custom tables/data, it would be neat to be able to save an instance of that specific Whoops generated page, and later be able to load it as a view to keep records of errors without only having the stack trace like the JsonResponseHandler outputs. I explored the idea of saving an instance of the class on runtime to the database, but the data is not finalized until after the \Whoops\Handler\PrettyPageHandler->handler() call so you can't save a serialized version of the class as the data is not loaded at that point.

TLDR; It would be a cool feature to save an instance of all the data that the PrettyPageHandler has, including custom tables and variables, and then later load it into the view.

**Comment by denis-sokolov** [comment]
This is a nice feature, but, alas, we are very far from supporting it, and I don’t expect us to be able to. Thank you for a thorough description, nevertheless!
